packages/web_ui/src/components/ModsPage.tsx

```diff
@@ -129,6 +136,403 @@ function CreateModPackButton() {
+					Close
+				</Button>,
+				// Remove explicit search button as form updates trigger search/filter
+			]}
+		>
+			<Form
+				form={form}
+				layout="horizontal"
+				labelCol={{ span: 6 }}
+				wrapperCol={{ span: 18 }}
+				onValuesChange={handleSearch}
+				initialValues={{ factorioVersion: factorioVersion, name: searchText }}
+			>
+				<Form.Item name="name" label="Filter by Name or Title">
+					<Input placeholder="Start typing to filter..." />
+				</Form.Item>
+				<Form.Item
+					name="factorioVersion"
+					label="Factorio Version"
+				>
+					<Select>
+						{/* Map allowed versions to Select Options */}
+						<Select.Option value="2.0">2.0</Select.Option>
+						<Select.Option value="1.1">1.1</Select.Option>
+						<Select.Option value="1.0">1.0</Select.Option>
+						<Select.Option value="0.18">0.18</Select.Option>
+						<Select.Option value="0.17">0.17</Select.Option>
+						<Select.Option value="0.16">0.16</Select.Option>
+						<Select.Option value="0.15">0.15</Select.Option>
+						<Select.Option value="0.14">0.14</Select.Option>
+						<Select.Option value="0.13">0.13</Select.Option>
+					</Select>
+				</Form.Item>
+			</Form>
+
+			{/* Display error message if fetching failed */}
+			{error && <Alert
+				message="Error Fetching Mods"
+				description={error.message}
+				type="error"
+				showIcon
+				style={{ marginBottom: 16 }}
+			/>}
+
+			<Table
+				dataSource={displayedMods}
+				rowKey={record => record.name}
+				loading={loading}
+				onChange={handleTableChange}
+				pagination={{
+					current: page,
+					pageSize: pageSize,
+					total: totalResults,
+					showSizeChanger: true,
+					pageSizeOptions: ["10", "20", "50", "100"],
+				}}
+				expandable={{
+					expandedRowRender: record => (
+						<div>
+							<p><strong>Summary:</strong> {record.summary ?? "N/A"}</p>
+							<p><strong>Downloads:</strong> {record.downloads_count?.toLocaleString() ?? "N/A"}</p>
+							<p><strong>Latest Release:</strong></p>
+							{record.latest_release ? (
+								<ul>
+									<li>Version: {record.latest_release.version ?? "N/A"}</li>
+									<li>
+										Factorio Version: {
+											record.latest_release.info_json?.factorio_version ?? "N/A"
+										}
+									</li>
+									<li>
+										Released: {
+											record.latest_release.released_at
+												? new Date(record.latest_release.released_at).toLocaleString()
+												: "N/A"
+										}
+									</li>
+								</ul>
+							) : <p>No release information available.</p>}
+							{/* Download button logic remains the same, assuming backend handles downloads */}
+							{account.hasPermission("core.mod.download") && record.latest_release && (
+								<Button
+									onClick={() => {
+										handleControllerDownload(
+											record.name,
+											record.title,
+											record.latest_release?.version,
+											factorioVersion
+										);
+									}}
+									disabled={!record.latest_release?.version}
```

Comment:
`record.latest_release?` can be `record.latest_release!` here because of the check on line 435 `&& record.latest_release`